cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/keyspaces/KeyspaceCreator.java

```diff
@@ -77,26 +78,43 @@ public CassandraStorageSettings getSettings()
 	}
 	
 	
-	public void createKeyspace() throws IOException
+	public void createKeyspace() throws IOException, CradleStorageException
 	{
 		Optional<KeyspaceMetadata> meta = obtainKeyspaceMetadata();
-		if (!meta.isPresent())
-		{
-			logger.info("Creating keyspace '{}'", keyspace);
-			CreateKeyspace createKs = settings.getNetworkTopologyStrategy() != null 
-					? SchemaBuilder.createKeyspace(keyspace).withNetworkTopologyStrategy(settings.getNetworkTopologyStrategy().asMap()) 
-					: SchemaBuilder.createKeyspace(keyspace).withSimpleStrategy(settings.getKeyspaceReplicationFactor());
-			queryExecutor.executeQuery(createKs.asCql(), true);
-			logger.info("Keyspace '{}' has been created", keyspace);
-			this.keyspaceMetadata = obtainKeyspaceMetadata().get();  //FIXME: keyspace creation may take time and it won't be available immediately
-		}
-		else
+		if (meta.isPresent())
+			throw new CradleStorageException("Keyspace '" + keyspace + "' already exists");
+
+		logger.info("Creating keyspace '{}'", keyspace);
+		CreateKeyspace createKs = settings.getNetworkTopologyStrategy() != null
+				? SchemaBuilder.createKeyspace(keyspace).withNetworkTopologyStrategy(settings.getNetworkTopologyStrategy().asMap())
+				: SchemaBuilder.createKeyspace(keyspace).withSimpleStrategy(settings.getKeyspaceReplicationFactor());
+		queryExecutor.executeQuery(createKs.asCql(), true);
+		logger.info("Keyspace '{}' has been created", keyspace);
+
+		awaitKeyspaceReady();
+	}
+
+	private void awaitKeyspaceReady() throws CradleStorageException
+	{
+		int timeout = 500;
```

Comment:
I would move timeouts/attempts (or max timeout and calculating this timeout automatically) to configuration.
Also I would add logging (may be debug) to see how much time did it actually sleep

Reply:
Oh, you're right! This should be configurable.

Reply:
What if I will use an existing timeout from CassandraStorageSettings and number of attempts will be hardcoded?